### ISSUE TITLE
lint for redundant trait bounds (due to trait inheritance)

### ISSUE THREAD

**Issue by thestinger** [rust-lang/rust#9197]
We should give a warning that the `Foo` bound is redundant:

~~~rust
trait Foo {}
trait Bar: Foo {}

fn foo<T: Foo + Bar>() {}
~~~

**Comment by tamird** [comment]
This should either be closed or the E-needstest label removed. Here's a case that demonstrates the problem:

```rust
trait Foo {
  fn noop(&self) {}
}

trait Bar: Foo {}

#[allow(dead_code)]
fn bar<T: Foo + Bar>() {}

fn main() {}
```

**Comment by frewsxcv** [comment]
It looks like there is no error nor warning for redundant traits

http://is.gd/iqumGL

**Comment by steveklabnik** [comment]
Since new lints have a big impact on users of rustc, the policy is that they should go through the RFC process like other user-facing changes. As such, I'm going to give this one a close, but if anyone comes across this ticket and wants this lint, consider adding it to clippy and/or writing up an RFC. Thanks!
